sr_robot_commander/test/test_robot_commander.py

```diff
@@ -0,0 +1,465 @@
+                 'ra_shoulder_lift_joint': -0.9569080092786892,
+                 'ra_wrist_3_joint': -0.25991215955733704}
+        rs = RobotState()
+        for key, value in state.items():
+            rs.joint_state.name.append(key)
+            rs.joint_state.position.append(value)
+        pose = self.robot_commander.get_end_effector_pose_from_state(rs)
+
+    def test_mvgr_get_planning_frame(self):
+        raised = False
+        try:
+            self.robot_commander.get_planning_frame()
+        except (NameError, TypeError):
+            raised = True
+        self.assertFalse(raised)
+
+    def test_mvgr_set_and_get_pose_reference_frame(self):
+        reference_frame = "world"
+        self.robot_commander._move_group_commander.set_pose_reference_frame(reference_frame)
+        self.assertTrue(self.robot_commander._move_group_commander.get_pose_reference_frame() == reference_frame)
+
+    def test_get_group_name(self):
+        self.assertTrue(self.robot_commander._name == self.robot_commander.get_group_name())
+
+    def test_refresh_named_targets(self):
+        self.robot_commander.refresh_named_targets()
+        condition_1 = type(self.robot_commander._srdf_names) == list
+        condition_2 = type(self.robot_commander._warehouse_names) == list
+        self.assertTrue(condition_1 and condition_2)
+
+    def test_set_max_velocity_scaling_factor__range_ok(self):
+        raised = False
+        try:
+            self.robot_commander.set_max_velocity_scaling_factor(0.2)
+        except Exception:
+            raised = True
+        self.assertFalse(raised)
+
+    def test_set_max_velocity_scaling_factor__range_not_ok(self):
+        raised = False
+        try:
+            self.robot_commander.set_max_velocity_scaling_factor(3)
+        except Exception:
+            raised = True
+        self.assertTrue(raised)
+
+    def test_set_max_acceleration_scaling_factor__range_ok(self):
+        raised = False
+        try:
+            self.robot_commander.set_max_acceleration_scaling_factor(0.2)
+        except Exception:
+            raised = True
+        self.assertFalse(raised)
+
+    def test_set_max_acceleration_scaling_factor__range_not_ok(self):
+        raised = False
+        try:
+            self.robot_commander.set_max_acceleration_scaling_factor(3)
+        except Exception:
+            raised = True
+        self.assertTrue(raised)
+
+    def test_allow_looking(self):
+        self.robot_commander.allow_looking(True)
+
+    def test_allow_replanning(self):
+        self.robot_commander.allow_replanning(True)
+
+    def test_plan_to_joint_value_target(self):
+        arm_home_joints_goal = {'ra_shoulder_pan_joint': 0.00, 'ra_elbow_joint': 2.00,
+                                'ra_shoulder_lift_joint': -1.25, 'ra_wrist_1_joint': -0.733,
+                                'ra_wrist_2_joint': 1.5708, 'ra_wrist_3_joint': 0.00}
+        self.robot_commander.plan_to_joint_value_target(arm_home_joints_goal,
```

Comment:
this function returns a plan, we should use the return of the function to check. I think it would be good to check if the last point of the plan corresponds with the goal you set, more than just seeing if it's a robot_trajectory message.

Reply:
Done